src/libs/Navigation/AppNavigator/createSearchFullscreenNavigator/index.tsx

```diff
@@ -4,18 +4,32 @@ import usePreserveNavigatorState from '@libs/Navigation/AppNavigator/createSplit
 import useNavigationResetOnLayoutChange from '@libs/Navigation/AppNavigator/useNavigationResetOnLayoutChange';
 import createPlatformStackNavigatorComponent from '@navigation/PlatformStackNavigation/createPlatformStackNavigatorComponent';
 import defaultPlatformStackScreenOptions from '@navigation/PlatformStackNavigation/defaultPlatformStackScreenOptions';
-import type {CustomEffectsHookProps, PlatformStackNavigationEventMap, PlatformStackNavigationOptions, PlatformStackNavigationState} from '@navigation/PlatformStackNavigation/types';
+import type {
+    CustomEffectsHookProps,
+    CustomStateHookProps,
+    PlatformStackNavigationEventMap,
+    PlatformStackNavigationOptions,
+    PlatformStackNavigationState,
+} from '@navigation/PlatformStackNavigation/types';
 import SearchFullscreenRouter from './SearchFullscreenRouter';
 
 function useCustomEffects(props: CustomEffectsHookProps) {
     useNavigationResetOnLayoutChange(props);
     usePreserveNavigatorState(props.state, props.parentRoute);
 }
 
+// This is a custom state hook that is used to render the last two routes in the stack.
+// We do this to improve the performance of the search results screen.
```

Comment:
If we use this for other navigators too, should we abstract it out and share it?

Reply:
@adamgrzybowski once you can get to this

Reply:
We do it in split navigators and here. The general rule is the same, but the version for split navigators looks different. The hook for split navigators is shared amongst them

Reply:
To clarify why they are different

SplitNavigators -> Sidebar that can be a separate route in the state
SearchNavigator -> Sidebar is part of other screens. It's not a separate route.

The difference is that the user can't open this screen by it's own on the narrow layout 
<img width="414" alt="image" src="https://github.com/user-attachments/assets/ca3e2fc5-023a-42cd-b1ba-358247d2dfe5" />